src/common/RWRef.h

```diff
@@ -0,0 +1,173 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2020 Xiubo Li <xiubli@redhat.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software 
+ * Foundation.  See file COPYING.
+ *
+ */
+
```

Comment:
Please add a a nice clear comment here about what a RWRef is intended to represent and how it works. In what situations would I want to use this new infrastructure?

Reply:
Sure. Will do it.

Reply:
Nice explanation.